apps/server/src/modules/learnroom/controller/rooms.controller.spec.ts

```diff
@@ -0,0 +1,85 @@
+import { Test, TestingModule } from '@nestjs/testing';
+import { EntityId, ICurrentUser } from '@shared/domain';
+import { BoardMapper } from '../mapper/board.mapper';
+import { RoomsUc, Board } from '../uc/rooms.uc';
+import { BoardResponse } from './dto/roomBoardResponse';
+import { RoomsController } from './rooms.controller';
+
+describe('rooms controller', () => {
+	let controller: RoomsController;
+	let mapper: BoardMapper;
+	let uc: RoomsUc;
+
+	beforeEach(async () => {
+		const module: TestingModule = await Test.createTestingModule({
+			imports: [],
+			providers: [
+				RoomsController,
+				{
+					provide: RoomsUc,
+					useValue: {
+						// eslint-disable-next-line @typescript-eslint/no-unused-vars
+						getBoard(roomId: EntityId, userId: EntityId): Promise<Board> {
+							throw new Error('please write mock for RoomsUc.getBoard');
+						},
+					},
+				},
+				{
+					provide: BoardMapper,
+					useValue: {
+						// eslint-disable-next-line @typescript-eslint/no-unused-vars
+						mapToResponse(board: Board): BoardResponse {
+							throw new Error('please write mock for Boardmapper.mapToResponse');
+						},
+					},
+				},
+			],
+		}).compile();
+		controller = module.get(RoomsController);
+		mapper = module.get(BoardMapper);
+		uc = module.get(RoomsUc);
+	});
+
+	describe('getContent', () => {
+		describe('when simple room is fetched', () => {
+			const setup = () => {
+				const currentUser = { userId: 'userId' } as ICurrentUser;
+
+				const ucResult = { roomId: 'id', title: 'title', displayColor: '#FFFFFF', elements: [] } as Board;
```

Comment:
what about a test with filled task within 'elements'?

Reply:
I think its not needed, because it doesnt matter to the controller. He simply passes the entire room into the mapper, tasks or no tasks, and returns whatever the mapper returns, tasks or no tasks.